src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -868,8 +870,17 @@ FixedwingPositionControl::control_position(const Vector2f &curr_pos, const Vecto
 		} else if (pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
 
 			/* waypoint is a loiter waypoint */
-			_l1_control.navigate_loiter(curr_wp, curr_pos, pos_sp_curr.loiter_radius,
-						    pos_sp_curr.loiter_direction, nav_speed_2d);
+			float loiter_radius = pos_sp_curr.loiter_radius;
+			uint8_t loiter_direction = pos_sp_curr.loiter_direction;
+
+			if (pos_sp_curr.loiter_radius <= 0) { //TODO: Get loiter radius from parameter
+				loiter_radius = _parameters.loiter_radius;
+				loiter_direction = 1;
```

Comment:
You could maybe have the direction be set depending on the sign of the radius, similar to [here](https://github.com/PX4/Firmware/blob/cf8f03f190d787d19225704b4badc508dcf4797e/src/modules/navigator/mission_block.cpp#L526).

Reply:
Thanks, was not aware having lotier radius as a negative value was possible. Addressed on last commit